.github/workflows/build_test.yml

```diff
@@ -189,3 +167,140 @@ jobs:
           cache-from: type=registry,ref=ghcr.io/cyclus/cyclus_rocky_${{ matrix.rocky_versions }}/cyclus:ci-layer-cache
           build-args: |
             rocky_version=${{ matrix.rocky_versions }}
+
+
+  build-test-macos:
```

Comment:
Thoughts on having these different platform builds in different workflows?  We've done that in other projects, but I'm not sure what a good practice is?  One benefit is that they can run in parallel.... but not sure of other benefits or downsides.

Reply:
- As is (multiple jobs in the same workflow) the jobs do run in parallel.  They only run serially if you specify dependencies using the `needs` key
- I like the idea of separating concerns and having different platforms in different workflows.  That feels like better practice to me.  The only significant change I can think of is that [this comment](https://github.com/cyclus/cyclus/pull/1743#issuecomment-2092064703) will need to be split apart - one for each platform

Reply:
Should I pull Rocky out to its own workflow or leave it in `build_test.yml`?  (can be a separate PR)